pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -222,10 +222,15 @@ func (rc *reconciler) reconcile() {
 					klog.Warningf(attachedVolume.GenerateMsgDetailed("attacherDetacher.DetachVolume started", fmt.Sprintf("This volume is not safe to detach, but maxWaitForUnmountDuration %v expired, force detaching", rc.maxWaitForUnmountDuration)))
 				}
 			}
-			if err != nil && !exponentialbackoff.IsExponentialBackoff(err) {
-				// Ignore exponentialbackoff.IsExponentialBackoff errors, they are expected.
-				// Log all other errors.
-				klog.Errorf(attachedVolume.GenerateErrorDetailed("attacherDetacher.DetachVolume failed to start", err).Error())
+			if err != nil {
```

Comment:
wouldn't code in https://github.com/kubernetes/kubernetes/pull/104526/files#diff-e76cff1d139f1df6d8c37cea61b19aa8caea6b103d64e58a084f2d81c5bad149L159 prevent detach operation from starting if there was another operation in-progress for the volume?

Reply:
Sorry it is not part of the diff but I was referring to code in:

```
if rc.attacherDetacher.IsOperationPending(attachedVolume.VolumeName, "" /* podName */, "" /* nodeName */) {
       klog.V(10).Infof("Operation for volume %q is already running in the cluster. Can't start detach for %q", attachedVolume.VolumeName, attachedVolume.NodeName)
	continue
}
```

Reply:
With the new changes (IsOperationSafeToRetry), the ASW attach state will not be modified 1. if there is a pending operation 2. or there is a backoff condition that is set.
Adding back the volume to the attached state for all error condition (in line 227) looks good to me.

Note:
With the old check of IsOperationPending() it does not handle the backoff condition and would remove the volume for ASW, hence the bug.

Reply:
ack. I see how this will fix the case of pending in exp. backoff.